### ISSUE TITLE
crash at startup

### ISSUE THREAD

**Issue by shizunge** [0xERR0R/blocky#849]
I encountered a crash, soon after startup.

I have not found how to reproduce it consistently. I guess it may be related to setting `blocking.startStrategy` to `fast` and `startVerifyUpstream` to false.

Here is the stack:
```
2023-01-28 14:28:18	panic: runtime error: invalid memory address or nil pointer dereference
2023-01-28 14:28:18	[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xcbbe38]
2023-01-28 14:28:18	
2023-01-28 14:28:18	goroutine 2294 [running]:
2023-01-28 14:28:18	github.com/0xERR0R/blocky/resolver.addExtraReasoning(0xc0002ae230?)
2023-01-28 14:28:18		/go/src/resolver/ede_resolver.go:42 +0x18
2023-01-28 14:28:18	github.com/0xERR0R/blocky/resolver.(*EdeResolver).Resolve(0xc004518810, 0xc000087b40?)
2023-01-28 14:28:18		/go/src/resolver/ede_resolver.go:24 +0x48
2023-01-28 14:28:18	github.com/0xERR0R/blocky/resolver.(*ClientNamesResolver).Resolve(0xc000080320, 0xc000225110)
2023-01-28 14:28:18		/go/src/resolver/client_names_resolver.go:79 +0x18c
2023-01-28 14:28:18	github.com/0xERR0R/blocky/resolver.(*FqdnOnlyResolver).Resolve(0x10?, 0x10?)
2023-01-28 14:28:18		/go/src/resolver/fqdn_only_resolver.go:34 +0xf4
2023-01-28 14:28:18	github.com/0xERR0R/blocky/resolver.(*FilteringResolver).Resolve(0xc0045187e0, 0xc000225110)
2023-01-28 14:28:18		/go/src/resolver/filtering_resolver.go:29 +0x1bc
2023-01-28 14:28:18	github.com/0xERR0R/blocky/server.(*Server).OnRequest(0xc0001eb590, {0x108b978, 0xc000182800}, 0xc0002350e0)
2023-01-28 14:28:18		/go/src/server/server.go:617 +0xee
2023-01-28 14:28:18	github.com/miekg/dns.HandlerFunc.ServeDNS(0xc0000c25a0?, {0x108b978?, 0xc000182800?}, 0xc00023001c?)
2023-01-28 14:28:18		/go/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:37 +0x2f
2023-01-28 14:28:18	github.com/miekg/dns.(*ServeMux).ServeDNS(0xc00032a280?, {0x108b978?, 0xc000182800?}, 0xc0002350e0?)
2023-01-28 14:28:18		/go/pkg/mod/github.com/miekg/dns@v1.1.50/serve_mux.go:103 +0x72
2023-01-28 14:28:18	github.com/miekg/dns.(*Server).serveDNS(0xc0000c90e0, {0xc00032a280, 0x36, 0x36}, 0xc000182800)
2023-01-28 14:28:18		/go/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:659 +0x462
2023-01-28 14:28:18	github.com/miekg/dns.(*Server).serveTCPConn(0xc0000c90e0, 0xc000030f90?, {0x108bb88?, 0xc000012450})
2023-01-28 14:28:18		/go/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:571 +0x2ce
2023-01-28 14:28:18	created by github.com/miekg/dns.(*Server).serveTCP
2023-01-28 14:28:18		/go/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:472 +0x2c5
```

**Comment by shizunge** [comment]
Seems related to https://github.com/0xERR0R/blocky/discussions/776

**Comment by kwitsch** [comment]
Seems like a problem with the ede resolver.
Extra information can't be added to the record.

Could you provide an example config to reproduce the problem?

**Comment by kwitsch** [comment]
Fixed in development (#778).
Duplicate report (#777 & #776).
